lib/widgets/context_menu/context_sub_menu_entry.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:files/widgets/context_menu/context_menu.dart';
+import 'package:files/widgets/context_menu/context_menu_entry.dart';
+import 'package:flutter/material.dart';
+
+/// [ContextSubMenuEntry] is a [PopupMenuEntry] that displays a submenu with [entries].
+class ContextSubMenuEntry extends BaseContextMenuEntry {
+  /// The [entries] are displayed in the order they are provided.
+  /// They can be [ContextMenuEntry], [ContextMenuDivider], [ContextSubMenuEntry] or any other widgets inherited from PopupMenuEntry.
+  final List<PopupMenuEntry> entries;
+
+  const ContextSubMenuEntry({
+    required String id,
+    required Widget title,
+    required this.entries,
+    Widget? leading,
+    bool enabled = true,
+    Key? key,
+  }) : super(
+          id: id,
+          leading: leading,
+          title: title,
+          enabled: enabled,
+          key: key,
+        );
+
+  @override
+  _ContextSubMenuEntryState createState() => _ContextSubMenuEntryState();
+
+  @override
+  double get height => 40;
+
+  @override
+  bool represents(String? value) => id == value;
+}
+
+class _ContextSubMenuEntryState extends State<ContextSubMenuEntry> {
+  @override
+  Widget build(BuildContext context) {
+    return InkWell(
+      onTap: widget.enabled
+          ? () {
+              // Get current render box of the context widget (ContextSubMenuEntry).
+              final RenderBox renderBox =
+                  context.findRenderObject()! as RenderBox;
+              // Get the position where the submenu should be opened.
+              final Offset offset =
+                  renderBox.localToGlobal(Offset(renderBox.size.width + 1, -8));
+
+              openContextMenu(context, offset, widget.entries);
+            }
+          : null,
+      child: Container(
+        height: widget.height,
+        padding: const EdgeInsets.symmetric(horizontal: 16),
+        child: Row(
+          children: [
+            if (widget.leading != null) ...[
+              IconTheme.merge(
+                data: IconThemeData(
+                  size: 20,
+                  color: widget.enabled
+                      ? Theme.of(context).colorScheme.onSurface.withOpacity(0.7)
```

Comment:
consider having the ternary operator inside the withOpacity call just to switch 0.7/0.3? Mostly a nit for code readability, could be wrong too, test and see if it works

Reply:
To avoid hardcoded variables in context\_menu\_entry.dart we could use ThemeExtension's, and control all parameters context menu related in this extension. Something like this:

```dart
class ContextMenuTheme extends ThemeExtension<ContextMenuTheme> {
  const ContextMenuTheme({
    required this.background,
    required this.textColor,
    this.enabledTextOpacity = 0.7,
    this.disabledTextOpacity = 0.3,
    this.enabledShortcutOpacity = 0.5,
    this.disabledShortcutOpacity = 0.1,
  });
```

Inside ThemeData:

```dart
    extensions: const [
      ContextMenuTheme(
        background: Color(0xFF212121),
        textColor: Colors.white,
      )
    ],
```

Inside context\_menu\_entry.dart

```dart
      color: theme.textColor.withOpacity(
        widget.enabled
            ? theme.enabledTextOpacity
            : theme.disabledTextOpacity,
      ),
```

Where 
```dart
theme = Theme.of(context).extension<ContextMenuTheme>()!;
```

Reply:
The extension is something that would be appreciated yea, only nit don't name the variable theme eventually, just call it something like menuTheme or smth